### DISCUSSION TITLE
Why not to use useState+useEffect instead useSyncExternalStore to support concurrency mode?

### DISCUSSION THREAD

**Discussion by budarin** [pmndrs/zustand#2318] [Q&A] [answered]
In useValito you wtote
>However, the "Sync" behavior doesn't work nicely with concurrent rendering. We can't use startTransition as expected.

Why not to use useState+useEffect in Zustand too instead useSyncExternalStore to support concurrency mode?

**Comment by dai-shi** [comment, marked as answer]
Because we care more about tearing than transition / suspense.
React might improve uSES. https://github.com/facebook/react/issues/26814
Otherwise, we may consider a different approach in the future version of Zustand.

In the meantime, feel free to try [use-zustand](https://github.com/dai-shi/use-zustand).

**Comment by budarin** [comment]
What do you mean by the term "tearing"?

Unfortunately, the discussions do not show either the intentions of the React team or a sane way to fix this.

**Reply by dai-shi** [reply]
Here is some links: https://github.com/dai-shi/will-this-react-global-state-work-in-concurrent-rendering?tab=readme-ov-file#what-is-tearing

**Reply by budarin** [reply]
Yeah, I came across this when I used the View Transitions API - React "lost" the state while doing the prettiness

Loss of state is a critical issue in the application
